packages/docusaurus-plugin-content-docs/src/sidebars/generator.ts

```diff
@@ -218,8 +218,8 @@ export const DefaultSidebarItemsGenerator: SidebarItemsGenerator = async ({
 
       function getCategoryLinkedDocId(): string | undefined {
         const link = categoryMetadata?.link;
-        if (link) {
-          if (link.type === 'doc') {
+        if (link !== undefined) {
```

Comment:
nice catch

can you add a test for this?

We already have one that you could use for inspiration:


```tsx
  test('uses explicit link over the index/readme.{md,mdx} naming convention', async () => {
    mockCategoryMetadataFiles({
      'Category/_category_.yml': {
        label: 'Category label',
        link: null
      },
    });
//...
```